ExadelTimeTrackingSystem/ExadelTimeTrackingSystem.Data/Repositories/ProjectRepository.cs

```diff
@@ -39,5 +39,21 @@ public Task<string> GetNameAsync(Guid id, CancellationToken cancellationToken)
             var filter = filterBuilder.Eq(p => p.Id, id);
             return GetCollection<Project>().Find(filter).Project(p => p.Name).SingleOrDefaultAsync();
         }
+
+        public async Task<bool> ActivityExistsAsync(Guid id, string activity, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            var filterBuilder = Builders<Project>.Filter;
+            var filter = filterBuilder.Eq(p => p.Id, id);
+            var project = await GetCollection<Project>().Find(filter).SingleOrDefaultAsync();
```

Comment:
Bad, because you are loading all heavy project model which can contain 100 fields. You can check existence of Activity inside the mongo with special method of filter builder, so that you should load **no** data of project at all. Find this necessary method, it's analog of Contains but in mongo